homeassistant/components/opower/coordinator.py

```diff
@@ -86,6 +87,35 @@ async def _async_update_data(
         # Because Opower provides historical usage/cost with a delay of a couple of days
         # we need to insert data into statistics.
         await self._insert_statistics()
+        # Estimate forecast values with the last known rate if they are empty
+        for _, forecast in enumerate(forecasts):
+            if (
+                forecast.usage_to_date > 0
+                and forecast.cost_to_date == 0
+                and forecast.account.utility_account_id in self.last_known_rates
+            ):
+                forecast.cost_to_date = (
+                    self.last_known_rates[forecast.account.utility_account_id]
```

Comment:
Use uuid for key. There is a separate effort to switch to uuid instead of utility_account_id to allow multiple meters under the same account.